node/modules/storageminer_idxprov.go

```diff
@@ -23,25 +24,44 @@ type IdxProv struct {
 	Datastore dtypes.MetadataDS
 }
 
-func IndexProvider(cfg config.IndexProviderConfig) func(params IdxProv, marketHost host.Host, dt dtypes.ProviderDataTransfer, maddr dtypes.MinerAddress) (provider.Interface, error) {
-	return func(args IdxProv, marketHost host.Host, dt dtypes.ProviderDataTransfer, maddr dtypes.MinerAddress) (provider.Interface, error) {
+func IndexProvider(cfg config.IndexProviderConfig) func(params IdxProv, marketHost host.Host, dt dtypes.ProviderDataTransfer, maddr dtypes.MinerAddress, ps *pubsub.PubSub) (provider.Interface, error) {
+	return func(args IdxProv, marketHost host.Host, dt dtypes.ProviderDataTransfer, maddr dtypes.MinerAddress, ps *pubsub.PubSub) (provider.Interface, error) {
 		ipds := namespace.Wrap(args.Datastore, datastore.NewKey("/index-provider"))
 		var opts = []engine.Option{
 			engine.WithDatastore(ipds),
 			engine.WithHost(marketHost),
 			engine.WithRetrievalAddrs(marketHost.Addrs()...),
+			engine.WithEntriesCacheCapacity(cfg.EntriesCacheCapacity),
+			engine.WithEntriesChunkSize(cfg.EntriesChunkSize),
+			engine.WithTopicName(cfg.TopicName),
+			engine.WithPurgeCacheOnStart(cfg.PurgeCacheOnStart),
```

Comment:
were these config options just not being passed in before?

Reply:
Nope; I must have missed them. They were all set to default

Reply:
Moved to [a separate PR](https://github.com/filecoin-project/lotus/pull/8444) to push that fix though without having to wait on this PR.